src/url-replacements.js

```diff
@@ -239,6 +242,14 @@ class UrlReplacements {
       return this.getTimingData_('navigationStart',
           'domContentLoadedEventStart');
     });
+
+    // Returns an identifier for the viewer.
+    this.set_('VIEWER', () => {
+      const viewerUrl = viewerFor(this.win_).getParam(VIEWPORT_ORIGIN_PARAM_) ||
```

Comment:
Checking for `origin` in the URL is not a great idea because it's very easily spoofable. That's why we have `messagingOrigin_` which is set via handshake. It all depends on the sensitivity of the problem. For instance, we have both: unconfirmed and confirmed referrers. The reason is because spoofing a referrer is not always a risk. How much risk here is to spoof viewer origin for your needs?